File: SafeCityAPI/src/main/java/com/ph/dao/KeyDaoImpl.java
package com.ph.dao;

import com.ph.model.Key;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.mongodb.core.MongoOperations;
import org.springframework.data.mongodb.core.query.Criteria;
import org.springframework.data.mongodb.core.query.Query;
import org.springframework.stereotype.Repository;


/**
 * Created by leon on 24/05/2017.
 */
@Repository("keyDao")
public class KeyDaoImpl implements KeyDao{

    @Autowired
    private MongoOperations mongoOperations;

    @Override
    public Key getGoogleMapsApiKey() {
        Query searchQuery = new Query(Criteria.where("name").is("google.api"));
        searchQuery.limit(1);
        return mongoOperations.findOne(searchQuery, Key.class);
    }



}

File: SafeCityAPI/src/main/java/com/ph/dao/KeyDaoImpl.kt
package com.ph.dao

import com.ph.model.Key
import com.ph.service.FireStoreServiceImpl
import org.springframework.stereotype.Service
import com.google.cloud.firestore.DocumentSnapshot
import org.springframework.core.convert.TypeDescriptor.collection
import com.google.cloud.firestore.QuerySnapshot
import com.google.api.core.ApiFuture




@Service("keyDao")
class KeyDaoImpl(/*private val mongoOperations: MongoOperations*/ private val fireStoreService: FireStoreServiceImpl) : KeyDao {

    override fun getGoogleMapsApiKey(): Key {
        val future = fireStoreService.db.collection("keys").whereEqualTo("name", "google.api").get()
        val documents = future.get().documents
        return documents[0].toObject(Key::class.java)
//        val searchQuery = Query(Criteria.where("name").`is`("google.api"))
//        searchQuery.limit(1)
//        return mongoOperations.findOne(searchQuery, Key::class.java)
    }
}